### PR TITLE
Rename barclamp nova_dashboard to horizon

### PR DESCRIPTION
**Pull request by vwallfahrer**
Depends on https://github.com/crowbar/crowbar-openstack/pull/21

### CODE REVIEW

**Comment by vuntz** [comment]
Unfortunately, you need to keep the nova_dashboard bits for released versions of the product. So this should use horizon for Cloud 6+ and nova_dashboard for earlier versions.

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2487,10 +2487,10 @@ function get_novacontroller()
 
 function get_novadashboard()
```
Comment (rsalevsky) [inline scripts/qa_crowbarsetup.sh]: This method needs some renaming too.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -2450,7 +2460,13 @@ function onadmin_proposal()
 
     prepare_proposals
 
-    local proposals="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova nova_dashboard ceilometer heat manila trove tempest"
+    local proposals="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova ceilometer heat manila trove tempest"
+
+    if [[ iscloudver 6plus ]]; then
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: This does not work with [[ ]] - just leave it out.

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -2518,14 +2534,22 @@ function get_novacontroller()
...
-function get_novadashboard()
+function get_horizon()
 {
-    local element=`crowbar nova_dashboard proposal show default | \
-        rubyjsonparse "
-                    puts j['deployment']['nova_dashboard']\
-                        ['elements']['nova_dashboard-server']"`
-    novadashboardserver=`resolve_element_to_hostname "$element"`
-    novadashboardservice=`resolve_element_to_hostname "$element" service`
+    if [[ iscloudver 6plus ]]; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: invalid syntax

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -3216,6 +3240,14 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
     local barclamp=""
     local proposal=""
     local applied_proposals=""
+    local all_barclamps="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova horizon ceilometer heat trove tempest"
+
+    if [[ iscloudver 6plus ]]; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: invalid syntax

**Comment by dirkmueller** [comment]
I think the rename of novadashboard to horizon (e.g. the shell variables and the function names) should / could be done in a separate commit (or even separate PR)

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -3216,6 +3240,14 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
     local barclamp=""
     local proposal=""
     local applied_proposals=""
+    local all_barclamps="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova horizon ceilometer heat trove tempest"
+
+    if iscloudver 6plus; then
+	all_barclamps = "$all_barclamps horizon"
+    else
+	all_barclamps = "$all_barclamps nova_dashboard"
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: No space around `=`.

File: scripts/qa_crowbarsetup.sh [hunk 6]
```diff
@@ -3224,7 +3256,7 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
     waitnodes nodes
 
     # reenable and apply the openstack propsals
-    for barclamp in pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova nova_dashboard ceilometer heat trove tempest ; do
+    for barclamp in  barclamps; do
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: I guess you meant `$all_barclamps`?

File: scripts/qa_crowbarsetup.sh [hunk 7]
```diff
@@ -3378,8 +3410,16 @@ function onadmin_teardown()
...
     local service
-    for service in nova_dashboard nova glance ceph swift keystone database ; do
+
+    if iscloudver 6plus; then
+	all_services = "$all_services horizon"
+    else
+	all_services = "$all_services nova_dashboard"
+    fi
+
+    for service in all_services ; do
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: Missing `$`.

File: scripts/qa_crowbarsetup.sh [hunk 8]
```diff
@@ -3378,8 +3410,16 @@ function onadmin_teardown()
...
 
     # undo propsal create+commit
+    local all_services = "nova glance ceph swift keystone database"
     local service
-    for service in nova_dashboard nova glance ceph swift keystone database ; do
+
+    if iscloudver 6plus; then
+	all_services = "$all_services horizon"
+    else
+	all_services = "$all_services nova_dashboard"
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: No space around `=`.

File: scripts/qa_crowbarsetup.sh [hunk 9]
```diff
@@ -2450,7 +2460,13 @@ function onadmin_proposal()
 
     prepare_proposals
 
-    local proposals="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova nova_dashboard ceilometer heat manila trove tempest"
+    local proposals="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova ceilometer heat manila trove tempest"
+
+    if iscloudver 6plus; then
+	proposals = " $proposals horizon"
+    else
+       proposals = "$proposals nova_dashboard"
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: No space around `=` and no space before `$proposals`.

File: scripts/qa_crowbarsetup.sh [hunk 10]
```diff
@@ -1788,6 +1788,10 @@ function enable_ssl_generic()
             $p "$a['ssl']['enabled']" true
             $p "$a['novnc']['ssl']['enabled']" true
         ;;
+        horizon)
+            $p "$a['apache']['ssl']" true
+            return
+        ;;
         nova_dashboard)
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: horizon|nova_dashboard)

File: scripts/qa_crowbarsetup.sh [hunk 11]
```diff
@@ -2027,6 +2032,11 @@ function custom_configuration()
                 proposal_set_value nova default "['attributes']['nova']['use_shared_instance_storage']" "true"
             fi
         ;;
+        horizon)
+            if [[ $hacloud = 1 ]] ; then
+                proposal_set_value horizon default "['deployment']['horizon']['elements']['horizon-server']" "['cluster:$clusternameservices']"
+            fi
+        ;;
         nova_dashboard)
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: horizon|nova_dashboard)
use "$proposal"

File: scripts/qa_crowbarsetup.sh [hunk 12]
```diff
@@ -2524,14 +2540,22 @@ function get_novacontroller()
...
+    if iscloudver 6plus ; then
+        local element=`crowbar horizon proposal show default | \
+            rubyjsonparse "
+                        puts j['deployment']['horizon']\
+                            ['elements']['horizon-server']"`
+    else
+        local element=`crowbar nova_dashboard proposal show default | \
+            rubyjsonparse "
+                        puts j['deployment']['nova_dashboard']\
+                            ['elements']['nova_dashboard-server']"`
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: use $var for horizon/nova_dashboard

File: scripts/qa_crowbarsetup.sh [hunk 13]
```diff
@@ -3222,6 +3246,14 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
     local barclamp=""
     local proposal=""
     local applied_proposals=""
+    local all_barclamps="pacemaker database rabbitmq keystone swift ceph glance cinder neutron nova horizon ceilometer heat trove tempest"
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: dup horizon, order changed

**Comment by bmwiedemann** [comment]
:+1: reviewed - still needs test

**Comment by bmwiedemann** [comment]
test failed on cloud6, because there is only a nova_dashboard barclamp:
```
+ crowbar horizon proposal create default
Usage: crowbar <area> <subcommand>
  Areas: batch ceilometer ceph cinder crowbar database deployer dns glance heat ipmi keystone logging machines manila network neutron nfs_client node_state nova nova_dashboard ntp pacemaker provisioner rabbitmq reset reset_nodes reset_proposal suse_manager_client swift tempest trove updater
```

**Comment by vwallfahrer** [comment]
Test fails because the other PR (https://github.com/crowbar/crowbar-openstack/pull/21) isn't merged, yet.

**Comment by bmwiedemann** [comment]
:+1: made jenkins test it together with https://github.com/crowbar/crowbar-openstack/pull/21 and it only failed in tempest (likely unrelated).

**Comment by rsalevsky** [comment]
+1

**Comment by vuntz** [comment]
Merging with the other PR this depends on.

**Pull request merged**